### PR TITLE
Feat/provide custom key builder

### PR DESCRIPTION
**Pull request by half-shell**

### CODE REVIEW

**Comment by half-shell** [comment]
I dropped the `package-lock.json` file since we get a warning because we might be using 2 package managers, which is not great. Was there a reason for us to have both a `package-lock.json` and a `yarn.lock`?

File: package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "test:watch": "jest --watch"
   },
   "dependencies": {
-    "express": "4.17.1",
+    "express": "4.17.2",
     "lodash": "4.17.21",
     "redis": "2.8.0"
```
Comment (rimiti) [inline package.json]: Could you please update Redis to v3 (at less).

File: package.json [hunk 2]
```diff
@@ -38,7 +38,7 @@
     "test:watch": "jest --watch"
   },
   "dependencies": {
-    "express": "4.17.1",
+    "express": "4.17.2",
```
Comment (rimiti) [inline package.json]: If express is only used for the typing, it seems not necessary for the package usage.

File: package.json [hunk 3]
```diff
@@ -10,18 +10,18 @@
     "access": "public"
   },
   "devDependencies": {
-    "@types/express": "4.17.11",
-    "@types/jest": "26.0.23",
-    "@types/lodash": "4.14.168",
     "@types/redis": "2.8.28",
```
Comment (rimiti) [inline package.json]: Can updated to v3.x (at less)

File: package.json (hunk 1 above)
Comment (half-shell) [reply package.json]: I'll try again, but the somehow the API does not really like it

File: src/classes/apiCache/index.ts [hunk 4]
```diff
@@ -57,15 +61,21 @@ export class ApiCache {
    * @param {number} durationInMS Cache expiration in ms
    * @returns Promise<string>
```
Comment (rimiti) [inline src/classes/apiCache/index.ts]: boolean?

File: src/classes/apiCache/types.ts [hunk 5]
```diff
@@ -1,6 +1,36 @@
...
+/**
+ * @description Function to be used to build the redis key. It defaults to a function building a string from the prefix followed by
+ * 1. The method
+ * 2. The path
+ * 3. The URL parameters
+ * @param {Request} req An express request.
+ * @param {prefix} prefix The prefix provided in the apiCache constructor's configuration. Defaults to empty string ''.
+ * @returns {string} Returns a string used to write / fetch the data in redis.
+ */
+type TKeyBuilder = (req: Request, prefix: string) => string;
```
Comment (rimiti) [inline src/classes/apiCache/types.ts:21]: 🎖️

File: package.json (hunk 3 above)
Comment (half-shell) [reply package.json]: There is no version 3.X.X. That is the issue. Also, `@types/redis` is deprecated for the new redis versions

**Review by rimiti — APPROVED** [review]
Minor comments.

LGTM

File: src/classes/apiCache/index.ts (hunk 4 above)
Comment (half-shell) [reply src/classes/apiCache/index.ts]: No, `redis.get()` returns the exact REDIS output, which is `OK`. We keep getting this wrong in other places.

Comment (rimiti) [reply src/classes/apiCache/index.ts]: My remark was just to say that the method signature seems different than the jsdoc:

```ts
async setCache(req: Request, data: unknown, durationInMS: number = this.config.expirationInMS): Promise<boolean> {
```

```js
  /**
   * @description Stores data into cache.
   * @param {Request} req Express request associated with the data
   * @param {unknown} data - Data to cache
   * @param {number} durationInMS Cache expiration in ms
   * @returns Promise<string>
   */
```

Comment (half-shell) [reply src/classes/apiCache/index.ts]: Oh got it, i'll fix it

File: package.json (hunk 1 above)
Comment (rimiti) [reply package.json]: API, is already in v3 :)

**Pull request merged**